PXP 4889 Docs(gen3-features): Added Gen3 features page (#87)
uc-cdis/gen3.org: content/resources/gen3-features.md

---
identifier: gen3-features

> title: Gen3 features

Reviewer: ```suggestion
title: Gen3 Features
```

linktitle: /resources/gen3-features
date: 2019-12-04T10:57:21-05:00
---

# A list of Gen3 features

Gen3 is an open-source platform that enables the standing-up of data commons for managing, analyzing, and sharing research data. Below you can find an exhaustive list of Gen3 features.

## Dictionary and metadata

* Explore the graph in the dictionary page of Windmill:
    * Search keywords
    * Visualize possible paths between nodes
    * Download JSON or TSV format for node templates 

> * Explorer project graph and nodes in Windmill:

Reviewer: ```suggestion
* Explore project graphs and nodes in Windmill:
```

>     * Visulize graph model

Reviewer: ```suggestion
    * Visualize graph model
```

    * Browse/download data for each node

> * Metadata submission (create, update, delete records) in JSON, TSV or CSV:

Reviewer: Generally, I think we call "metadata" -> "structured data".  
I don't feel strongly about this, but I was told this a while back by Bob..

Author: Good point, replaced metadata with structured data

    * Submission page of Windmill
    * Gen3 SDK (Python, R)
* Export metadata as a JSON, TSV or CSV file
* The metadata is stored in Postgres databases
* ElasticSearch:
    * The metadata can be stored in ElasticSearch by using our ETL tool
    * Explore the ElasticSearch metadata using Guppy
* Generate fake data specific to a dictionary for testing
* Import and export using the Portable Format for Biomedical Data (PFB): *(Work in progress)*
    * Import and export a dictionary
    * Import and export metadata

## Data files

* Data upload through the Gen3 Client
* Indexing data files:
    * Create data file records and mint GUIDs
    * Index a new version of the same file
    * Get the most recent version of a file
    * Index data files that are stored outside of the Commons
* GUID resolver
* Data download:
    * Gen3 Client
    * Files page of Windmill

> * Core metadata endpoint to get information about a data file

Reviewer: Out of curiosity, what is this? Is this the indexd endpoint? Or something else?

Author: Yes, indexd endpoint

Reviewer 2: this is the pidgin endpoint

Reviewer 2: it should be called "core metadata" because that's what is used in the pidgin docs. it uses the core_metadata_collection node. so renaming it "core structured data" does not work

Author: Thank you for the comment, fixed in the next commit.

---
identifier: gen3-features
title: Gen3 Features
linktitle: /resources/gen3-features
layout: withtoc
menuname: featuresMenu
date: 2019-12-04T10:57:21-05:00
---

# A list of Gen3 features

Gen3 is an open-source platform that enables the standing-up of data commons for managing, analyzing, and sharing research data. Below you can find an exhaustive list of Gen3 features.

## Dictionary and structured data

* Explore the graph in the dictionary page of Windmill:
    * Search keywords
    * Visualize possible paths between nodes
    * Download JSON or TSV format for node templates 
* Explore project graphs and nodes in Windmill:
    * Visualize graph model
    * Browse/download data for each node
* Structured data submission (create, update, delete records) in JSON, TSV or CSV:
    * Submission page of Windmill
    * Gen3 SDK (Python, R)
* Export structured data as a JSON, TSV or CSV file
* Structured data is stored in Postgres databases
* ElasticSearch:

>     * Structured data can be stored in ElasticSearch by using our ETL tool

Reviewer 3: Just my thoughts:  this could change to `Structured data can be flattened and stored in ElasticSearch by using our ETL tool`, and then its parent bullet `ElasticSearch` could change to `flattened data`.  Sorry I should probably also change this in our wiki repo, I will do that today :)

Author: Thank you for the feedback, Qingya!

    * Explore the ElasticSearch structured data using Guppy
* Generate fake data specific to a dictionary for testing
* Import and export using the Portable Format for Biomedical Data (PFB): *(Work in progress)*
    * Import and export a dictionary
    * Import and export structured data

## Data files

* Data upload through the Gen3 Client
* Indexing data files:
    * Create data file records and mint GUIDs
    * Index a new version of the same file
    * Get the most recent version of a file
    * Index data files that are stored outside of the Commons
* GUID resolver
* Data download:
    * Gen3 Client
    * Files page of Windmill
* Core structured data endpoint to get information about a data file

## Data exploration and analysis

* GraphQL endpoint and UI to explore structured data via Peregrine (graph data)

> * GraphQL endpoint and UI to explore Structured data via Guppy (flat ElasticSearch data)

Reviewer 3: ```suggestion
* GraphQL endpoint and UI to explore flattened data via Guppy (in ElasticSearch)
```
